snuba/utils/streams/backends/abstract.py

```diff
@@ -34,6 +34,12 @@ def __init__(self, partition: Partition, offset: int):
         self.offset = offset
 
 
+class OffsetOutOfRange(ConsumerError):
+    """
+    Raised when trying to read from an invalid position in the partition.
+    """
```

Comment:
Kafka unfortunately doesn't provide any details about _what_ offset is invalid, so we're stuck with the lowest common denominator here.

It also seems to basically render the consumer unusable — all subsequent calls raise this exception:

```
cimpl.KafkaException: KafkaError{code=_STATE,val=-172,str="Failed to seek to offset 1: Local: Erroneous state"}
```

I did not try to replicate that behavior in the local storages.